sql/sql_table.cc

```diff
@@ -17502,7 +17502,6 @@ bool mysql_alter_table(THD *thd, const char *new_db, const char *new_name,
                                   &table_def))
       return true;
 
-    table_list->partition_names = &alter_info->partition_names;
```

Comment:
Unfortunately, we can't be sure that secondary_engine_unlock_table() doesn't use TABLE_LIST::partition_names because we don't have access to code of Oracle's Heatwave/Rapid SE which is called from this routine. Moreover, it is likely that it does use this info, even though it is rather awkward from SQL-layer/SE-isolation point of view, because I can see TABLE_LIST::partition_names being assigned in other places the code calls secondary load/unload code.

So I think that it is better to keep this line if we want to be compatible with Oracle.
Instead we can save original table_list->partition_names value on the stack and restore it after calling secondary_engine_unload().

Reply:
Interesting... the semantics of std::exchange() is not that obvious from its name...

I think it is OK to push the patch now.

Reply:
Yeah, I had the same impression when I saw this "exchange" name for the first time :)